From bde61c86ce9c33b40cee6f88a249c1c435445474 Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Thu, 11 Apr 2019 12:46:51 +0200
Subject: [PATCH] Revert "bus/policy: define the type of the policy
 serialization statically"

The preprocessor uses too much memory for this on some architectures.
Go back to doing the computation at run-time.

This reverts commit 7b2a8674c9cfd1f65b3e0097870a922b69f00172.
---
 src/bus/policy.c | 110 +----------------------------------------------
 1 file changed, 2 insertions(+), 108 deletions(-)

diff --git a/src/bus/policy.c b/src/bus/policy.c
index a4cf5a3..e8ae26f 100644
--- a/src/bus/policy.c
+++ b/src/bus/policy.c
@@ -3,7 +3,6 @@
  */
 
 #include <c-dvar.h>
-#include <c-dvar-type.h>
 #include <c-list.h>
 #include <c-rbtree.h>
 #include <c-stdaux.h>
@@ -15,112 +14,6 @@
 #include "util/error.h"
 #include "util/selinux.h"
 
-static const CDVarType policy_type[] = {
-        C_DVAR_T_INIT(
-                C_DVAR_T_TUPLE4(
-                        C_DVAR_T_ARRAY(
-                                C_DVAR_T_TUPLE2(
-                                        C_DVAR_T_u,
-                                        C_DVAR_T_TUPLE5(
-                                                C_DVAR_T_b,
-                                                C_DVAR_T_t,
-                                                C_DVAR_T_ARRAY(
-                                                        C_DVAR_T_TUPLE4(
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_s
-                                                        )
-                                                ),
-                                                C_DVAR_T_ARRAY(
-                                                        C_DVAR_T_TUPLE10(
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_t
-                                                        )
-                                                ),
-                                                C_DVAR_T_ARRAY(
-                                                        C_DVAR_T_TUPLE10(
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_t
-                                                        )
-                                                )
-                                        )
-                                )
-                        ),
-                        C_DVAR_T_ARRAY(
-                                C_DVAR_T_TUPLE4(
-                                        C_DVAR_T_b,
-                                        C_DVAR_T_u,
-                                        C_DVAR_T_u,
-                                        C_DVAR_T_TUPLE5(
-                                                C_DVAR_T_b,
-                                                C_DVAR_T_t,
-                                                C_DVAR_T_ARRAY(
-                                                        C_DVAR_T_TUPLE4(
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_s
-                                                        )
-                                                ),
-                                                C_DVAR_T_ARRAY(
-                                                        C_DVAR_T_TUPLE10(
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_t
-                                                        )
-                                                ),
-                                                C_DVAR_T_ARRAY(
-                                                        C_DVAR_T_TUPLE10(
-                                                                C_DVAR_T_b,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_s,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_u,
-                                                                C_DVAR_T_t,
-                                                                C_DVAR_T_t
-                                                        )
-                                                )
-                                        )
-                                )
-                        ),
-                        C_DVAR_T_ARRAY(
-                                C_DVAR_T_TUPLE2(
-                                        C_DVAR_T_s,
-                                        C_DVAR_T_s
-                                )
-                        ),
-                        C_DVAR_T_b
-                )
-        )
-};
-
 static PolicyXmit *policy_xmit_free(PolicyXmit *xmit) {
         if (!xmit)
                 return NULL;
@@ -645,7 +538,8 @@ int policy_registry_import(PolicyRegistry *registry, CDVar *v) {
         bool apparmor;
         int r;
 
-        c_dvar_read(v, "<(", policy_type);
+        /* XXX: provide the type */
+        c_dvar_read(v, "<(", NULL);
 
         c_dvar_read(v, "[");
 
-- 
2.20.1

